Cargo.toml

```diff
@@ -655,8 +655,8 @@ slog-scope = "4.4.0"
 slog-term = "2.9.1"
 socket2 = { version = "0.5.7", features = ["all"] }
 static_assertions = "1.1.0"
-strum = { version = "0.26.2", features = ["derive"] }
```

Comment:
This removal breaks cargo compilation:

```
error[E0432]: unresolved import `strum::EnumIter`
   --> rs/types/types/src/lib.rs:109:5
    |
109 | use strum::EnumIter;
    |     ^^^^^^^^^^^^^^^ no `EnumIter` in the root
    |
```


 The documentation of strum https://docs.rs/strum/0.26.3/strum/derive.EnumIter.html says the `drive` feature is required for `EnumIter`, which is used above.

I wonder why our CI didn't catch it? @basvandijk @rumenov

Reply:
Strange. I don't see ic-types being compiled in the cargo build:
https://github.com/dfinity/ic/actions/runs/11163355802/job/31030110610

Reply:
Strange, I see ic-types was successfully built on CI with cargo:
https://github.com/dfinity/ic/actions/runs/11163355802/job/31030110610#step:5:1853

Reply:
I think I know the problem. `Cargo.lock` wasn't updated in this PR which meant ic-types was built with the old strum version. We need to make sure this can't happen.

Reply:
We need to pass `--locked` when we [cargo build](https://github.com/dfinity/ic/blob/master/.github/workflows-source/ci-main.yml#L388).

Reply:
sorry @ninegua, I would have expected that CI catches this.

Most likely it is just a matter of re-generating Cargo.lock as Bas said since the bazel tests are passing.